packages/viz/src/lib/Layer.ts

```diff
@@ -115,24 +108,35 @@ export class Layer {
     return this._deckLayer;
   }
 
+  private async _getLayerProperties() {
+    const metadata = this._source.getMetadata();
+
+    const styleProps = this._style
+      ? this._style.getProperties(this._source)
+      : undefined;
+
+    const props = this._source.getProps();
+
+    return {
+      ...this._options,
+      ...props,
+      ...defaultStyles(metadata.geometryType),
+      ...styleProps,
+      type: MVTLayer
+    };
+  }
   /**
    * Replace a layer source
-   * @param previousSource source of the layer to be replaced
    */
-  private async _replaceLayer(previousSource: Source) {
-    const newLayer = await this._createDeckGLLayer();
+  private async _replaceLayer() {
```

Comment:
I like the idea of reusing the layer

Reply:
Reusing the layer is a bad performance practice from Deck.gl. Here's [the FAQ](https://deck.gl/#/documentation/developer-guide/using-layers?section=faq) that explains the why. So I would leave it like I had it to be better in terms of performance.

Reply:
No idea of the perf issues, but agree to keep it if performances get better